server/backend/api/app/jwt_validation.py

```diff
@@ -228,3 +230,29 @@ def authorize_by_app_id(
 def get_access_roles(claims: dict = Depends(authorize)):
     groups = claims[JWT_GROUPS_KEY]
     return groups
+
+
+def get_username_from_id_token(
+    id_token: str
+) -> dict:
+    # TODO: The ID Token should be validated
+    try:
+        unverified_claims = jwt.get_unverified_claims(id_token)
+    except jwt.JWTError:
+        raise HTTPException(
+            status_code=401,
+            detail={'code': ERROR_TOKEN_DECODE,
+                    'description':
+                    'Unable to decode ID token.'},
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+
+    if JWT_USERNAME_KEY not in unverified_claims:
+        raise HTTPException(
+            status_code=403,
+            detail={'code': ERROR_USERNAME_REQUIRED,
+                    'description': 'Required custom:idp_username in claim'},
```

Comment:
Maybe 'Requires custom:idp_username in claim'

Reply:
Thanks, correcting it